src/Action/EntrypointAction.php

```diff
@@ -21,15 +21,17 @@
  */
 final class EntrypointAction
 {
+    private $entrypoint;
     private $resourceNameCollectionFactory;
 
-    public function __construct(ResourceNameCollectionFactoryInterface $resourceNameCollection)
+    public function __construct(Entrypoint $entrypoint, ResourceNameCollectionFactoryInterface $resourceNameCollectionFactory)
     {
-        $this->resourceNameCollectionFactory = $resourceNameCollection;
+        $this->entrypoint = $entrypoint;
```

Comment:
IMO the entrypoint should be a value object, not a service. Don't inject it, just return in `__invoke`: `new Entrypoint(...)`.